crates/test-macro/src/lib.rs

```diff
@@ -34,23 +35,95 @@ pub fn wasm_bindgen_test(
     let mut leading_tokens = Vec::new();
     while let Some(token) = body.next() {
         leading_tokens.push(token.clone());
-        if let TokenTree::Ident(token) = token {
-            if token == "async" {
-                r#async = true;
+
+        match token {
+            // Parse a potential `should_panic` attribute
+            // Start by parsing the `#`
+            TokenTree::Punct(op) if op.as_char() == '#' => {
+                // We need to peek into the next token, save it before we continue.
+                let token = if let Some(token) = body.next() {
+                    leading_tokens.push(token.clone());
```

Comment:
This is commonly done in attribute proc-macros in my experience. There might be other proc-macros the user is using that want to read this attribute.

Reply:
I think you are right though, it is only commonly done with foreign attributes, you should always consume your own attributes.

Right now this is ours, because `#[test]` isn't used here. I'm researching what Rust has to say to why the leftover attribute works here.

Reply:
Yeah, couldn't really find anything, will consume it.

Considering that we are consuming it, I will now actually error on parsing instead of `continue`.

Reply:
Done.